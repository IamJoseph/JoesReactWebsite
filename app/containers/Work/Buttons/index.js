import React from 'react';
import PropTypes from 'prop-types';
import { Container, LI, Link } from './ButtonsStyled';

export default function Buttons({ active }) {
  return (
    <Container>
      <LI color="198" active={active === '/work'} topRadius>
        <Link to="/work">Jobs</Link>
      </LI>
      <LI color="183" active={active === '/work/projects'}>
        <Link to="/work/projects">Projects</Link>
      </LI>
      <LI color="155" active={active === '/work/current_site'}>
        <Link to="/work/current_site">Current Site</Link>
      </LI>
      <LI color="97" active={active === '/work/client_sites'}>
        <Link to="/work/client_sites">Client Sites</Link>
      </LI>
      <LI color="65" active={active === '/work/graphics'} bottomRadius boxShadow>
        <Link to="/work/graphics">Graphics</Link>
      </LI>
    </Container>
  );
}

Buttons.propTypes = {
  active: PropTypes.string.isRequired,
};
