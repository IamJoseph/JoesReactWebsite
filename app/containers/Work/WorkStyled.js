import styled from 'styled-components';
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';

export const Container = styled.section`
background-color: #e4e2d7;
border: 1.5px solid #c2cdd6;
display: flex;
height: 100%;
justify-content: center;
margin-left: 25px;
margin-right: 25px;
max-width: 1460px;
padding: 0 .5em;
width: 100%;
z-index: 1;
@media(min-width: 768px){
  padding: 0 1em;
}
@media(min-width: 1150px){
  padding: 0 1.87em;
}
`;

export const PageContainer = styled(CSSTransitionGroup)`
  background: linear-gradient(to bottom, #f9f9f1 50%,#edebe1 90%, #edebe1 100%);
  border: 1.5px solid #9fb4c6;
  border-radius: 0 5px 5px 5px;
  box-shadow: 2px 2px 5px #bdbcb5;
  display: flex;
  flex: 8;
  height: 85%;
  justify-content: center;
  margin-top: 85px;
  max-height: calc(100% - 100px);
  overflow-y: auto;
  padding: 1em;
`;
