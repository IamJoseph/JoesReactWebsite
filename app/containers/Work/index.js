/*
 *
 * Work
 *
 */

import React, { PureComponent } from 'react';
import Helmet from 'react-helmet';
import PropTypes from 'prop-types';
import { Switch } from 'react-router-dom';
import { Container, PageContainer } from './WorkStyled';
import Buttons from './Buttons';
import AsyncRoute from '../../routing/AsyncRoute';
import currentLoader from './Pages/Current/loader';
import clientLoader from './Pages/Client/loader';
import graphicsLoader from './Pages/Graphics/loader';
import defaultLoader from './Pages/Default/loader';
import projectsLoader from './Pages/Projects/loader';

export default class Work extends PureComponent { // eslint-disable-line react/prefer-stateless-function
  getSubRouteContent = (path) => {
    const { store } = this.context;
    switch (path) {
      case '/work/current_site':
        return currentLoader(store);
      case '/work/client_sites':
        return clientLoader(store);
      case '/work/projects':
        return projectsLoader(store);
      case '/work/graphics':
        return graphicsLoader(store);
      default:
      case '/work':
        return defaultLoader(store);
    }
  }

  render() {
    const { pathname } = this.props.location;
    return (
      <Container>
        <Helmet>
          <title>Work</title>
          <meta name="description" content="Past employment, projects, skills" />
        </Helmet>
        <Buttons active={pathname} />
        <PageContainer
          component="div"
          className="transition-group"
          transitionName="page"
          transitionEnterTimeout={2000}
          transitionLeave={false}
          transitionAppear
          transitionAppearTimeout={2000}
        >
          {/* transition needs rendered element on pageload */}
          <div style={{ width: '100%', height: '100%' }}>
            <Switch>
              <AsyncRoute path={pathname} load={this.getSubRouteContent(pathname)} />
            </Switch>
          </div>
        </PageContainer>
      </Container>
    );
  }
}

Work.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }),
};

Work.defaultProps = {
  location: {
    pathname: '',
  },
};

Work.contextTypes = {
  store: PropTypes.object,
};
