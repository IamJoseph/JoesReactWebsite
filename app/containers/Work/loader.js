/**
 * Asynchronously loads the components for WorkPage
 */

import { errorLoading } from 'utils/asyncInjectors';

export default () => (cb) => {
  import('containers/Work')
    .then(cb)
    .catch(errorLoading);
};
