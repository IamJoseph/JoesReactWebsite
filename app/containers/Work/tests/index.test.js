import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Work from '../index';

describe('<Work />', () => {
  it('should render correctly w/ work path', () => {
    const fakeProps = {
      location: {
        pathname: '/work',
      },
    };
    const wrapper = shallow(
      <Work {...fakeProps} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly w/ current_site path', () => {
    const fakeProps = {
      location: {
        pathname: '/work/current_site',
      },
    };
    const wrapper = shallow(
      <Work {...fakeProps} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly w/ client_sites path', () => {
    const fakeProps = {
      location: {
        pathname: '/work/client_sites',
      },
    };
    const wrapper = shallow(
      <Work {...fakeProps} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly w/ projects path', () => {
    const fakeProps = {
      location: {
        pathname: '/work/projects',
      },
    };
    const wrapper = shallow(
      <Work {...fakeProps} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly w/ graphics path', () => {
    const fakeProps = {
      location: {
        pathname: '/work/graphics',
      },
    };
    const wrapper = shallow(
      <Work {...fakeProps} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with default value', () => {
    const fakeProps = {
      location: {
        pathname: '',
      },
    };
    const wrapper = shallow(
      <Work {...fakeProps} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
