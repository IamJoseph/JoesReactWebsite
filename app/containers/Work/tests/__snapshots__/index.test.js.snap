// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Work /> should render correctly w/ client_sites path 1`] = `
<styled.section>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <title>
      Work
    </title>
    <meta
      content="Past employment, projects, skills"
      name="description"
    />
  </HelmetWrapper>
  <Buttons
    active="/work/client_sites"
  />
  <Styled(CSSTransitionGroup)
    className="transition-group"
    component="div"
    transitionAppear={true}
    transitionAppearTimeout={2000}
    transitionEnterTimeout={2000}
    transitionLeave={false}
    transitionName="page"
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Switch>
        <AsyncRoute
          load={[Function]}
          path="/work/client_sites"
        />
      </Switch>
    </div>
  </Styled(CSSTransitionGroup)>
</styled.section>
`;

exports[`<Work /> should render correctly w/ current_site path 1`] = `
<styled.section>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <title>
      Work
    </title>
    <meta
      content="Past employment, projects, skills"
      name="description"
    />
  </HelmetWrapper>
  <Buttons
    active="/work/current_site"
  />
  <Styled(CSSTransitionGroup)
    className="transition-group"
    component="div"
    transitionAppear={true}
    transitionAppearTimeout={2000}
    transitionEnterTimeout={2000}
    transitionLeave={false}
    transitionName="page"
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Switch>
        <AsyncRoute
          load={[Function]}
          path="/work/current_site"
        />
      </Switch>
    </div>
  </Styled(CSSTransitionGroup)>
</styled.section>
`;

exports[`<Work /> should render correctly w/ graphics path 1`] = `
<styled.section>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <title>
      Work
    </title>
    <meta
      content="Past employment, projects, skills"
      name="description"
    />
  </HelmetWrapper>
  <Buttons
    active="/work/graphics"
  />
  <Styled(CSSTransitionGroup)
    className="transition-group"
    component="div"
    transitionAppear={true}
    transitionAppearTimeout={2000}
    transitionEnterTimeout={2000}
    transitionLeave={false}
    transitionName="page"
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Switch>
        <AsyncRoute
          load={[Function]}
          path="/work/graphics"
        />
      </Switch>
    </div>
  </Styled(CSSTransitionGroup)>
</styled.section>
`;

exports[`<Work /> should render correctly w/ projects path 1`] = `
<styled.section>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <title>
      Work
    </title>
    <meta
      content="Past employment, projects, skills"
      name="description"
    />
  </HelmetWrapper>
  <Buttons
    active="/work/projects"
  />
  <Styled(CSSTransitionGroup)
    className="transition-group"
    component="div"
    transitionAppear={true}
    transitionAppearTimeout={2000}
    transitionEnterTimeout={2000}
    transitionLeave={false}
    transitionName="page"
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Switch>
        <AsyncRoute
          load={[Function]}
          path="/work/projects"
        />
      </Switch>
    </div>
  </Styled(CSSTransitionGroup)>
</styled.section>
`;

exports[`<Work /> should render correctly w/ work path 1`] = `
<styled.section>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <title>
      Work
    </title>
    <meta
      content="Past employment, projects, skills"
      name="description"
    />
  </HelmetWrapper>
  <Buttons
    active="/work"
  />
  <Styled(CSSTransitionGroup)
    className="transition-group"
    component="div"
    transitionAppear={true}
    transitionAppearTimeout={2000}
    transitionEnterTimeout={2000}
    transitionLeave={false}
    transitionName="page"
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Switch>
        <AsyncRoute
          load={[Function]}
          path="/work"
        />
      </Switch>
    </div>
  </Styled(CSSTransitionGroup)>
</styled.section>
`;

exports[`<Work /> should render correctly with default value 1`] = `
<styled.section>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <title>
      Work
    </title>
    <meta
      content="Past employment, projects, skills"
      name="description"
    />
  </HelmetWrapper>
  <Buttons
    active=""
  />
  <Styled(CSSTransitionGroup)
    className="transition-group"
    component="div"
    transitionAppear={true}
    transitionAppearTimeout={2000}
    transitionEnterTimeout={2000}
    transitionLeave={false}
    transitionName="page"
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Switch>
        <AsyncRoute
          load={[Function]}
          path=""
        />
      </Switch>
    </div>
  </Styled(CSSTransitionGroup)>
</styled.section>
`;
